pinot-core/src/main/java/org/apache/pinot/core/transport/TableRouteInfo.java

```diff
@@ -0,0 +1,198 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.transport;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.pinot.common.request.BrokerRequest;
+import org.apache.pinot.common.request.InstanceRequest;
+import org.apache.pinot.core.routing.ServerRouteInfo;
+import org.apache.pinot.core.routing.TimeBoundaryInfo;
+import org.apache.pinot.spi.config.table.TableConfig;
+
+
+/**
+ * Routing information for a table required to route plan fragments to servers.
+ * RequestMap is used to submit plan fragments to servers.
+ * Offline and realtime routing tables are used to keep track of which servers are executing the query.
+ * getUnavailableSegments() returns the segments that are not available when calculating the routing table.
+ * numPrunedSegmentsTotal() returns the number of segments that were pruned when calculating the routing table.
+ */
+public interface TableRouteInfo {
+
+  @Nullable
+  String getOfflineTableName();
+
+  @Nullable
+  String getRealtimeTableName();
+
+  /**
+   * Gets the broker request for the offline table, if available.
+   *
+   * @return the broker request for the offline table, or null if not available
+   */
+  @Nullable
+  BrokerRequest getOfflineBrokerRequest();
+
+  /**
+   * Gets the broker request for the realtime table, if available.
+   *
+   * @return the broker request for the realtime table, or null if not available
+   */
+  @Nullable
+  BrokerRequest getRealtimeBrokerRequest();
+
+  /**
+   * Gets the table config for the offline table, if available.
+   * @return the table config for the offline table, or null if not available
+   */
+  @Nullable
+  TableConfig getOfflineTableConfig();
+
+  /**
+   * Gets the table config for the realtime table, if available.
+   * @return the table config for the realtime table, or null if not available
+   */
+  @Nullable
+  TableConfig getRealtimeTableConfig();
```

Comment:
These 2 doesn't apply to logical table

Reply:
You are right. Only Query & Routing configs are required from table configs. Once https://github.com/apache/pinot/issues/15607 is implemented, we can only return only those configs. Is it OK to keep it for now ? I can add TODO if necessary.

Reply:
We can keep the intermediate code as needed. Just want to make sure we know the end state and a path moving there

Reply:
Sure. I am working on a response to this slack conversation: https://apache-pinot.slack.com/archives/C07TESEPS4A/p1745838450704719
That should resolve how to handle table configs.